core/src/main/java/org/opensearch/sql/executor/execution/QueryPlanFactory.java

```diff
@@ -76,10 +77,15 @@ public AbstractPlan create(
   /**
    * Creates a ContinuePaginatedPlan from a cursor.
    */
-  public AbstractPlan create(String cursor, boolean isExplain,
+  public AbstractPlan create(String cursor, boolean isExplain, boolean isCursorClose,
```

Comment:
would it be smarter to have an action or resource type for this?  Instead of isExplain, or isCursorClose (etc) we could have action is an enum of type: 
- SEARCH
- EXPLAIN
- CURSOR_CLOSE
- ...etc

Reply:
Alternatively, use separate functions for each type of action: 
- `createSearch`
- `createExplain`
- `createCursorClose`
- ...etc

Reply:
We already have
- `QueryPlan`
- `ExplainPlan`
- time for `CommandPlan`, methinks.